addons/mail/models/mail_message.py

```diff
@@ -218,6 +221,15 @@ def _compute_is_current_user_or_guest_author(self):
             else:
                 message.is_current_user_or_guest_author = False
 
+    def _compute_is_current_user_can_unlink(self):
+        """ Compute if the current user can unlink the message. """
+        self.is_current_user_can_unlink = False
+        for message in self:
+            if message.model == 'discuss.channel' and message.res_id:
+                channel = self.env['discuss.channel'].browse(message.res_id)
+                if channel and (channel.is_self_channel_owner or channel.is_self_channel_admin):
+                    message.is_current_user_can_unlink = True
```

Comment:
And as the compute is related to channel it should be in discuss folder either way.